### ISSUE TITLE
[v5]Memory leak in stack

### ISSUE THREAD

**Issue by genglei01** [react-navigation/react-navigation#7686]
**Current Behavior**

- What code are you running and what is happening?

1.I create a simple project with react-navigation(Stack),the project only includes two screens (Home/Detail).Xcode displays the memory is used around 32M when project launch in release 

2.Go to detail screen with **navigation.navigate('Detail1')**,the memory is used around 38M and it does not come back to 32M when going back to Home Screen , even I waited about half minutes.

3.The memory is used about  43M when I navigate to Detail and back over three-time. and It seems the memory is not released.

- Include a screenshot or video if it makes sense.
![image](https://user-images.githubusercontent.com/3293583/75510147-eecdab00-5a24-11ea-8274-dce0e1196402.png)

**Expected Behavior**

- What do you expect should be happening?
I think maybe something makes memory leak.
The memory should come down  as the startup

How to make the memory come down?

**How to reproduce**
You can get my project from  https://github.com/genglei01/rn_test.git
**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS or Android                 | iOS
|@react-native-community/masked-view| ^0.1.6
|    @react-navigation/bottom-tabs|^5.1.0
|   @react-navigation/native|^5.0.8
|  @react-navigation/stack|^5.1.0
|  react| 16.9.0 
| react-native|0.61.5
|  react-native-gesture-handler|^1.6.0
 |   react-native-reanimated|^1.7.0
|  react-native-safe-area-context|^0.7.3
| react-native-screens"|^2.1.0

**Comment by satya164** [comment]
43M isn't much. It's likely that garbage collection isn't triggered because it's not needed and as a result memory isn't freed. Does the memory usage keep growing till it goes out of memory?

**Comment by genglei01** [comment]
@satya164  yes, it keeps growing. The memory usage is over 200M in my real project.

**Comment by wilau2** [comment]
Warning in development mode as well: 
![image](https://user-images.githubusercontent.com/5473183/77554586-a6bb7000-6e8c-11ea-86d8-e635c7e5c889.png)

**Comment by satya164** [comment]
@genglei01 but does it go out of memory? 200M isn't a lot with devices with GBs of memories. it's important to know that if it's an actual leak or just garbage collection is delayed because it's not needed.

**Comment by satya164** [comment]
Closing since there's no response. Will re-open if you can confirm if this is causing an OOM exception.

**Comment by LogunovAV** [comment]
It is causing OOM exception. We have a project where user can navigate through vritual artworks, each screen have lots of images and the design side implement a shortcut for search through it.

Our flow look like this: home -> search -> choose search result -> view result -> click shortcut for search -> search -> choose search result -> view result.

It looks like react navigation doesn't clear if same screen was open. We have tried to use each navigation method like direct navigation with navigator.navigate or pop screens.

**Comment by ducdh-fabbi** [comment]
I have same issue for both Android and iOS, please reopen this.

"@react-navigation/bottom-tabs": "^5.11.2",
"@react-navigation/drawer": "^5.11.4",
"@react-navigation/native": "^5.8.10",
"@react-navigation/stack": "^5.12.8",
"react": "16.13.1",
"react-native": "0.63.3",
"react-native-gesture-handler": "^1.6.1",
"react-native-screens": "^2.16.1",

**Comment by LogunovAV** [comment]
@satya164 can you re-open it, please? It looks important for all of us. Our project is finished, but I am still watching the issue.

**Comment by gyl89** [comment]
Related to https://github.com/react-navigation/react-navigation/issues/7078 ?

**Comment by leecommamichael** [comment]
> @genglei01 but does it go out of memory? 200M isn't a lot with devices with GBs of memories. it's important to know that if it's an actual leak or just garbage collection is delayed because it's not needed.

This is an unfortunate perspective. I support devices that have 1 and 2 GB of RAM (iOS) at work.

**Comment by erik-kallen** [comment]
In my case, at least, the issue seems to have been caused by react-native-reanimated: https://github.com/software-mansion/react-native-reanimated/pull/3687

**Comment by vnahornyi** [comment]
Hello! Is there any solution? I encountered the same problem on v6 after expanding the app and increasing the number of elements with images
